lib/tools/system-commands.js

```diff
@@ -80,71 +60,55 @@ systemCallMethods.connect = async function connect () {
 
   let isStartupMonitorEnabled = true;
   try {
-    if (_.isEmpty(await getPids(COMPANION_PGREP_PATTERN(this.udid)))) {
-      const args = buildCompanionArgs(this.udid, this.companion);
-      const comp = new SubProcess(IDB_COMPANION_EXECUTABLE, args);
-      comp.on('output', (stdout, stderr) => {
-        if (isStartupMonitorEnabled && stderr) {
-          log.debug(`[${IDB_COMPANION_EXECUTABLE}] ${stderr}`);
-        }
-      });
-      try {
-        await comp.start(null, PROCESS_INIT_TIMEOUT);
-      } catch (e) {
-        throw new Error(`Cannot start ${IDB_COMPANION_EXECUTABLE}. Original error: ${e.message}`);
-      }
-      await B.delay(300);
-      if (!comp.isRunning) {
-        throw new Error(`${IDB_COMPANION_EXECUTABLE} has failed to start`);
-      }
-      log.info(`${IDB_COMPANION_EXECUTABLE} has been started with args ${JSON.stringify(args)}`);
-    } else {
-      log.info(`${IDB_COMPANION_EXECUTABLE} is already running for the device '${this.udid}'`);
-    }
-
-    const daemon = new SubProcess(IDB_EXECUTABLE, buildDaemonArgs(this.executable));
-    let daemonStderr = '';
-    daemon.on('output', (stdout, stderr) => {
-      if (isStartupMonitorEnabled && stderr) {
-        daemonStderr += `${stderr}\n`;
-      }
-    });
-    try {
-      await daemon.start(null, PROCESS_INIT_TIMEOUT);
-      await B.delay(300);
-    } catch (ign) {}
-    if (daemon.isRunning) {
-      log.info(`${IDB_EXECUTABLE} daemon is running on port ${this.port}`);
-    } else {
-      if (!daemonStderr.includes('address already in use')) {
-        throw new Error(`${IDB_EXECUTABLE} daemon has failed to start: ${daemonStderr}`);
-      }
-      log.info(`${this.port} is already in use. Assuming it is used by ${IDB_EXECUTABLE} daemon`);
-    }
-
-    const {stdout} = await tpExec(IDB_EXECUTABLE, ['list-targets', '--json']);
-    if (!fixOutputToArray(stdout).some((x) => _.toUpper(x.udid) === _.toUpper(this.udid))) {
-      throw new Error(`The device '${this.udid}' is unknown. Only these devices are known ` +
-        `to ${IDB_EXECUTABLE}: ${stdout}`);
-    }
-
     try {
       await waitForCondition(async () => {
         try {
+          const daemon = new SubProcess(IDB_EXECUTABLE, buildDaemonArgs(this.executable));
+          let daemonOutput = '';
+          daemon.on('output', (stdout, stderr) => {
+            if (isStartupMonitorEnabled && _.trim(stdout || stderr)) {
+              daemonOutput += `[daemon] ${stdout || stderr}\n`;
+            }
+          });
+          try {
+            await daemon.start(null, PROCESS_INIT_TIMEOUT);
```

Comment:
No need for the `null` (https://github.com/appium/node-teen_process/blob/master/lib/subprocess.js#L56-L59)

Reply:
nope, this is exactly what I need:

```
if (startDetector === null) {
   startDetector = genericStartDetector;
}
```

instead of `startDetector = null`, which is going to hardcode the wait duration.